tests/unit/igraph_hsbm_game.c

```diff
@@ -0,0 +1,74 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+#include "test_utilities.h"
+
+void call_and_print(igraph_integer_t n, igraph_integer_t m, igraph_vector_t *rho, igraph_matrix_t *pref_matrix, igraph_real_t p) {
+    igraph_t result;
+    IGRAPH_ASSERT(igraph_hsbm_game(&result, n, m, rho, pref_matrix, p) == IGRAPH_SUCCESS);
```

Comment:
It is not necessary to assert that the return code is `IGRAPH_SUCCESS`. If it is not, the program aborts with an error and the test fails, even before the `IGRAPH_ASSERT` code could be reached.

We used to use this when the error handler was set to "ignore" so we could test invalid input. But a while ago you added a convenient macro to check bad input, so we almost never need to use a non-aborting error handler anymore.